pkg/compact/compact.go

```diff
@@ -966,16 +971,28 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 		}
 
 		// Send all groups found during this pass to the compaction workers.
+	groupLoop:
 		for _, g := range groups {
 			select {
-			case <-errGroupCtx.Done():
-				break
+			case err = <-errChan:
+				if err != nil {
```

Comment:
minor inconsistency. You are checking if err != nil here but not on line 992. I would say, don't check it our code couple of lines above never pushes nil? However if channel is closed you would get nil here, but reader is responsible for closing

Reply:
Yeah good point.  I've removed the extra check here as I don't think it's needed, as you said the reader is doing the closing.